deploy/000025-create-test-network.ts

```diff
@@ -1,49 +1,132 @@
-import { PoolType } from '../utils/Constants';
-import { deploy, DeployedContracts, execute, InstanceName, isLive, setDeploymentMetadata } from '../utils/Deploy';
+import { MAX_UINT256, PoolType } from '../utils/Constants';
+import {
+    deploy,
+    DeployedContracts,
+    execute,
+    InstanceName,
+    isLive,
+    setDeploymentMetadata,
+    TestTokenInstanceName
+} from '../utils/Deploy';
 import { duration } from '../utils/Time';
 import { DEFAULT_DECIMALS, TokenData, TokenSymbol } from '../utils/TokenData';
 import { toPPM, toWei } from '../utils/Types';
+import { BigNumberish } from 'ethers';
 import { DeployFunction } from 'hardhat-deploy/types';
 import { HardhatRuntimeEnvironment } from 'hardhat/types';
 
-const INITIAL_SUPPLY = toWei(1_000_000_000);
-
-const DEPOSIT_LIMIT = toWei(5_000_000);
-const FUNDING_LIMIT = toWei(10_000_000);
 const TRADING_FEE = toPPM(0.2);
 const BNT_VIRTUAL_BALANCE = 1;
 const BASE_TOKEN_VIRTUAL_BALANCE = 2;
 
-const INITIAL_DEPOSITS = {
-    [InstanceName.TestToken1]: toWei(50_000),
-    [InstanceName.TestToken2]: toWei(500_000),
-    [InstanceName.TestToken3]: toWei(1_000_000),
-    [InstanceName.TestToken4]: toWei(2_000_000),
-    [InstanceName.TestToken5]: toWei(3_000_000),
-    [InstanceName.TestToken6]: toWei(100_000, new TokenData(TokenSymbol.TKN6).decimals())
+interface TokenPoolConfig {
+    symbol: TokenSymbol;
+    instanceName: TestTokenInstanceName;
+    initialSupply: BigNumberish;
+    initialDeposit: BigNumberish;
+    depositLimit: BigNumberish;
+    fundingLimit: BigNumberish;
+    tradingDisabled?: boolean;
+    depositingDisabled?: boolean;
+}
+
+const normalizeAmounts = (config: TokenPoolConfig) => {
+    const decimals = new TokenData(config.symbol).decimals();
+
+    return {
+        symbol: config.symbol,
+        instanceName: config.instanceName,
+        initialSupply: toWei(config.initialSupply, decimals),
+        initialDeposit: toWei(config.initialDeposit, decimals),
+        depositLimit: config.depositLimit !== MAX_UINT256 ? toWei(config.depositLimit, decimals) : MAX_UINT256,
+        fundingLimit: config.fundingLimit !== MAX_UINT256 ? toWei(config.fundingLimit, decimals) : MAX_UINT256,
```

Comment:
I'm pretty sure that you need to convert to BigNumber and then use `.eq` (otherwise, the comparison will always return false).
Also, instead of a comparison, you could simply do the more generic assignment:
```
depositLimit: BigNumber.min(toWei(config.depositLimit, decimals), MAX_UINT256)
```

Reply:
Please note that since the values are being instantiated by the same `MAX_UINT256` object (i.e., not a value), `===` will work just as well. In any case, I'll change to `BigNumber.min`.